Controller/Adminhtml/Importcategory/Save.php

```diff
@@ -71,147 +79,191 @@ public function __construct(
                     $store = $this->_storeManager->getStore();
                     $storeId = $store->getStoreId();
-                    $singlestoremode = $this->_storeManager->isSingleStoreMode();
+                    $singleStoreMode = $this->_storeManager->isSingleStoreMode();
                     $_stores = [];
-                    if (!$singlestoremode) {
+                    
+                    if (!$singleStoreMode) {
                         $stores = $this->_storeManager->getStores();
+                        
                         foreach ($stores as $key => $store) {
                             $_stores[$store->getCode()] = $store->getId();
                         }
                     }
+                    
                     $rootNodeId = $store->getRootCategoryId();
                     $rootCat = $this->_categoryFactory->create();
-                    $cat_info = $rootCat->load($rootNodeId);
-
-                        $alreadyexist = [];
-                        $categorycollection = $this->_categoryFactory->create()->getCollection()->addAttributeToSelect('name');
-                        $exist_categories_name = [];
-                        $exist_categories_path = [];
-                        $exist_categories_pathname = [];
-                    foreach ($categorycollection as $key => $value) {
-                        $exist_categories_name[$value->getId()] = $value->getName();
-                        $exist_categories_path[$value->getId()] = $value->getPath();
-                        $checkcat = $this->_categoryFactory->create();
-                        $categoryobj = $checkcat->load($value->getId());
-                        $parentcatnames = [];
-                        $parentid = '';
-                        foreach ($this->getparentCategories($categoryobj) as $key => $parentcate) {
-                            $parentcatnames[] = $parentcate->getName();
-                            $parentid = $parentcate->getId();
+                    $categoryInfo = $rootCat->load($rootNodeId);
+
+                    $alreadyExist = [];
+                    $categoryCollection = $this->_categoryFactory
+                        ->create()
+                        ->getCollection();
+                    $existingCatPath = [];
+                    $existingCatPathName = [];
+                        
+                    foreach ($categoryCollection as $key => $value) {
+                        $existingCategories[$value->getId()] = $value->getName();
+                        $existingCatPath[$value->getId()] = $value->getPath();
+                        $checkCat = $this->_categoryFactory->create();
+                        $categoryObj = $checkCat->load($value->getId());
+                        $parentCatNames = [];
+                        $parentId = '';
+                        
+                        foreach ($this->getParentCategories($categoryObj) as $key => $parentCate) {
```

Comment:
can you use either `$parentCat` or `$parentCategory`

Reply:
![image](https://user-images.githubusercontent.com/22006850/45302970-9cd15680-b50c-11e8-9a05-eb3392b37acf.png)